src/app/containers/PageHandlers/withData.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { shape, element } from 'prop-types';
+import articlePropTypes from '../../models/propTypes/article';
+import ErrorMain from '../ErrorMain';
+import nodeLogger from '../../helpers/logger.node';
+import deepGet from '../../helpers/json/deepGet';
+
+const logger = nodeLogger(__filename);
+
+// checks for data and status keys, setting default status if not found
+const constructRenderObject = data => ({
+  status: deepGet(['status'], data) ? data.status : 500,
+  assetData: deepGet(['data'], data),
+});
+
+// checks for assetData and 200 status
+const shouldRender = data => {
+  let hasDataAnd200Status = false;
+  const { status, assetData } = constructRenderObject(data);
+
+  if (assetData && status === 200) hasDataAnd200Status = true;
+
+  return { hasDataAnd200Status, status, assetData };
+};
+
+const WithData = Component => {
+  const DataContainer = ({ data, ...props }) => {
+    const { hasDataAnd200Status, status, assetData } = shouldRender(data);
+    if (hasDataAnd200Status) {
+      try {
+        return <Component assetData={assetData} {...props} />;
+      } catch (err) {
+        logger.error(err);
+      }
+    }
+
+    DataContainer.propTypes = {
```

Comment:
I think these prop types should be inline with the component, not inside it

Reply:
`const WithData = Component => {
  const DataContainer = ({ data, ...props }) => {
    const { hasDataAnd200Status, status, assetData } = shouldRender(data);
    if (hasDataAnd200Status) {
      try {
        return <Component assetData={assetData} {...props} />;
      } catch (err) {
        logger.error(err);
      }
    }

    return <ErrorMain status={status} />;
  };

  DataContainer.propTypes = {
    data: shape(articlePropTypes),
  };

  DataContainer.defaultProps = {
    data: shape,
  };

  return DataContainer;
};`

Is this what you are suggesting @dr3

Reply:
Yeah, you shouldnt reference DataContainer from inside itself :)

Reply:
yeah, good spot @dr3